api/envoy/api/v2/cds.proto

```diff
@@ -581,6 +581,10 @@ message Cluster {
   // If this flag is not set to true, Envoy will wait until the hosts fail active health
   // checking before removing it from the cluster.
   bool drain_connections_on_host_removal = 32;
+
+  // If enabled, allow HTTP/gRPC requests to this cluster to use a hedging
+  // strategy in which multiple upstream requests may be sent simultaneously.
+  bool allow_request_hedging = 39;
```

Comment:
Why not just put this on the route? It doesn't seem like we need a cluster level config for this? It seems like the route/vhost config could be defined such that if the message is present it turns it on, even though it might not happen without headers?

Reply:
I did that in response to your previous comment:

> It might be sufficient in this case to make it a router stat vs. cluster stat. The other option is to only allow header override if the feature is configured in some way, then create the stats in a new sub-scope. Personally I think I prefer the latter option?

I could be wrong here, but I was thinking that the decision to allocate memory for cluster stats has to happen when the cluster is initialized, hence why I put this on the cluster. I'm requiring it up front because if I don't in this PR someone might enable it via route and not via the cluster and then be using the feature without stats.

Reply:
Hmm yeah fair enough. I guess I was thinking that we could dynamically allocate the stats the first time they are used on the route/cluster, but that might be a bit difficult unless we make them router stats and not cluster stats? It seems not great to have things like this hanging off of the cluster but I can't think of a better option in this case. WDYT about at least putting this inside `core.HttpProtocolOptions`? Any other ideas?

Reply:
If we don't want this option maybe I should just remove it for now? I think the best way would be to add "lazy" stats that only are initialized when first incremented, then we don't need config options or to use more memory in the common case.

Reply:
I'm still not sure how you would add lazy stats without making the cluster code more complicated. It's almost like we want cluster scoped router specific stats, but that is also complicated from a cleanup perspective. @envoyproxy/maintainers any thoughts on this one? @jmarantz in particular for comment?

Reply:
It seems to me like we're in a problematic spot where we used to just enable stats by default and that we're now in a situation where we find ourselves adding flags scattered around the code base to control whether to enable features to keep the number of stats under control. Maybe instead there should be be a way to centrally control stats - kinda like the existing `StatsMatcher` except that we default new stats to not being included. That way we could keep stats under control without having to add in feature specific config whenever new stats are added

Given how complicated this stats question is turning out, how do we feel about landing this change without stats and figuring out the best way of handling stats separately? I know we'd love to see this feature land and while stats would be nice we could get by without them for the time being

Reply:
I've come to the same conclusion as @snowp. I think we should probably trim down the cluster stats by default with a default exclusion set, and then people can-opt in, and we can document. Yes I think it's fine landing this without stats and then figuring it out. +1